### DISCUSSION TITLE
how do you access "use history"?

### DISCUSSION THREAD

**Discussion by kurakurakantor** [vrtmrz/obsidian-livesync#66] [Q&A] [answered]
sometimes my devices don't sync correctly and the older empty file win and rewrite the written file, so i think i should just turn on "use history" to be able to restore accidental deletes like that. but i can't seem to figure out how to see the history dialog. is it database only so you have to access the database itself manually? and another question, is the use history local only or also synced to the cloud database? and one more question, how to delete the history? i thought collect garbage would also delete the history but the history persists even after the garbage was collected

**Comment by kurakurakantor** [comment]
ah! figured it out! it's via the command palette! but i'm still wondering if it is local only or synced with the database? (and how to delete the history)

**Comment by vrtmrz** [comment, marked as answer]
This feature shows local changeset. So you can see changes that happened on your devices. 
But, if you enabled LiveSync, thinned remote changesets will be stored locally.

Garbage collection is a complicated feature. Excuse me for less documentation.
LiveSync's GC deletes old, not used chunks. It means marking `_deleted` in PouchDB/CouchDB.
In CouchDB, marked chunks will be left for a while. Otherwise, the Replication protocol will not able to know whether the chunk had been deleted or not replicated yet.
(When after compaction had run, it will be changed to `tombstone`. But, still not be deleted)

The reliable way to delete history on whole databases is running `Drop History` -> `Drop and Send` / `Drop and receive` on the Setting dialog.
Only about local device's history, hitting `Reset local database` and replicate again.

**Reply by kurakurakantor** [reply]
thank you so much for your answer!! i understand now
